### ISSUE TITLE
The size of the model on the GPU does not increase, but decreases with a decrease in model_size_reg < 0.5

### ISSUE THREAD

**Issue by manmitya** [catboost/catboost#1762]
Problem: 
The documentation says that the model should decrease when the model_size_reg parameter is greater than 0.5 and increases. However, the model is also reduced when the model_size_reg parameter (< 0.5) is reduced.
The size of the model on the GPU does not increase, but decreases with a decrease in model_size_reg < 0.5
For example, a dataset with a size of 340k, divided by 80/20 percent. 84 categorical features.
We train the model on the GPU and CPU. The resulting dimensions of the model are shown in the table.

model_size_reg | GPU, Accuracy at   test_pool | GPU, Model Size   (cbm), MB | CPU, Accuracy at   test_pool | CPU, Model Size (cbm),   MB
-- | -- | -- | -- | --
0,001 | 0,9103 | 835 | 0,9103 | 1014
0,002 | 0,9104 | 907 | 0,9104 | 1064
0,005 | 0,9106 | 901 |   |  
0,01 | 0,9103 | 881 |   |  
0,02 | 0,9106 | 969 |   |  
0,05 | 0,9109 | 1 268 |   |  
0,12 | 0,9095 | 2 196 | 0,9100 | 433
0,25 | 0,9054 | 3 947 | 0,9094 | 271
0,50 | 0,8914 | 4 670 | 0,9086 | 179
None | 0,8904 | 4 626 |   |  
1,00 | 0,8673 | 2 366 |   |  
2,00 | 0,8430 | 1 649 |   |  
4,00 | 0,8309 | 1 597 |   |  

Doc Ref: https://catboost.ai/docs/concepts/python-reference_parameters-list.html?lang=en

**Other model parameters:**
num_of_iteration = 150
learning_rate = 0.05
max_ctr_complexity = 5
one_hot_max_size = 20
max_depth = 10

catboost version: 0.26
Operating System: Windows 10 x64
Python 3.8.8
Anaconda3-2021.05-Windows-x86_64.exe
CPU: Intel Core i5-8250U CPU @ 1.60GHz
GPU: NVIDIA GeForce MX150

**Comment by ek-ak** [comment]
Hello! We fixed this bug in version 1.0.0 )
